### ISSUE TITLE
Java runtime version unsupported following Rundeck update

### ISSUE THREAD

**Issue by Erlum** [rundeck/rundeck#8820]
Bug description : 
Following update of Rundeck to version 5.0.1.20240115 on a Rocky 8 system via the community repository, Rundeck does not run anymore due ton unsupported java runtime version

Rundeck version : update to 5.0.1.20240115
Install type : RPM (baseurl=https://packages.rundeck.com/pagerduty/rundeck/rpm_any/rpm_any/$basearch)
OS version : Rocky Linux 8.9
DB type : PostgreSQL

To reproduce : 
dnf update rundeck -y

Expected behaviour : 
Rundeck should be updated and able to run after the update, however the java runtime version is now 55, and the logs say Rundeck only support up to 52, thus Rundeck is now unable to run.

Additional context : 
Note : logs have been slightly obfuscated as they come from a production environment (yes, I know, this shouldn't be used in prod, we have a licensed cluster ready to host our workflows, but did not switch yet)

DNF logs : 
> root@RUNDECKHOST user_erlum# dnf update rundeck
> Last metadata expiration check: 0:00:27 ago on Wed 17 Jan 2024 01:00:55 PM CET.
> Dependencies resolved.
> ====================================================================================================
>  Package                         Architecture  Version                       Repository        Size
> ====================================================================================================
> Upgrading:
>  rundeck                         noarch        5.0.1.20240115-1              rundeck          265 M
> Installing dependencies:
>  java-11-openjdk-headless        x86_64        1:11.0.21.0.9-2.el8_8         appstream         42 M
> 
> Transaction Summary
> ====================================================================================================
> Install  1 Package
> Upgrade  1 Package
> 
> Total download size: 306 M
> Is this ok [y/N]: y
> Downloading Packages:
> (1/2): java-11-openjdk-headless-11.0.21.0.9-2.el8_8.x86_64.rpm       43 MB/s |  42 MB     00:00
> (2/2): rundeck-5.0.1.20240115-1.noarch.rpm                           27 MB/s | 265 MB     00:09
> ----------------------------------------------------------------------------------------------------
> Total                                                                31 MB/s | 306 MB     00:09
> Running transaction check
> Transaction check succeeded.
> Running transaction test
> Transaction test succeeded.
> Running transaction
>   Running scriptlet: java-11-openjdk-headless-1:11.0.21.0.9-2.el8_8.x86_64                      1/1
>   Preparing        :                                                                            1/1
>   Installing       : java-11-openjdk-headless-1:11.0.21.0.9-2.el8_8.x86_64                      1/3
>   Running scriptlet: java-11-openjdk-headless-1:11.0.21.0.9-2.el8_8.x86_64                      1/3
>   Running scriptlet: rundeck-5.0.1.20240115-1.noarch                                            2/3
>   Upgrading        : rundeck-5.0.1.20240115-1.noarch                                            2/3
> warning: /etc/rundeck/framework.properties created as /etc/rundeck/framework.properties.rpmnew
> warning: /etc/rundeck/profile created as /etc/rundeck/profile.rpmnew
> warning: /etc/rundeck/realm.properties created as /etc/rundeck/realm.properties.rpmnew
> warning: /etc/rundeck/rundeck-config.properties created as /etc/rundeck/rundeck-config.properties.rpmnew
> 
>   Running scriptlet: rundeck-5.0.1.20240115-1.noarch                                            2/3
>   Running scriptlet: rundeck-4.17.3.20231113-1.noarch                                           3/3
>   Cleanup          : rundeck-4.17.3.20231113-1.noarch                                           3/3
>   Running scriptlet: rundeck-4.17.3.20231113-1.noarch                                           3/3
>   Running scriptlet: java-11-openjdk-headless-1:11.0.21.0.9-2.el8_8.x86_64                      3/3
>   Running scriptlet: rundeck-4.17.3.20231113-1.noarch                                           3/3
>   Verifying        : java-11-openjdk-headless-1:11.0.21.0.9-2.el8_8.x86_64                      1/3
>   Verifying        : rundeck-5.0.1.20240115-1.noarch                                            2/3
>   Verifying        : rundeck-4.17.3.20231113-1.noarch                                           3/3
> 
> Upgraded:
>   rundeck-5.0.1.20240115-1.noarch
> Installed:
>   java-11-openjdk-headless-1:11.0.21.0.9-2.el8_8.x86_64
> 
> Complete!

rundeckd service status : 
> root@RUNDECKHOST user_erlum# systemctl status rundeckd -l
> ● rundeckd.service - SYSV: rundeckd, providing rundeckd
>    Loaded: loaded (/etc/rc.d/init.d/rundeckd; generated)
>    Active: failed (Result: exit-code) since Wed 2024-01-17 13:02:22 CET; 9min ago
>      Docs: man:systemd-sysv-generator(8)
>   Process: 5792 ExecStop=/etc/rc.d/init.d/rundeckd stop (code=exited, status=0/SUCCESS)
>   Process: 5817 ExecStart=/etc/rc.d/init.d/rundeckd start (code=exited, status=0/SUCCESS)
>  Main PID: 5829 (code=exited, status=1/FAILURE)
> 
> Jan 17 13:02:21 RUNDECKHOST systemd[1]: Starting SYSV: rundeckd, providing rundeckd...
> Jan 17 13:02:21 RUNDECKHOST rundeckd[5817]: Starting rundeckd: [  OK  ]
> Jan 17 13:02:21 RUNDECKHOST systemd[1]: Started SYSV: rundeckd, providing rundeckd.
> Jan 17 13:02:22 RUNDECKHOST systemd[1]: rundeckd.service: Main process exited, code=exited, status=1/FAILURE
> Jan 17 13:02:22 RUNDECKHOST systemd[1]: rundeckd.service: Failed with result 'exit-code'.

/var/log/rundeck/service.log

> Exception in thread "main" java.lang.UnsupportedClassVersionError: rundeckapp/Application has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
>         at java.lang.ClassLoader.defineClass1(Native Method)
>         at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
>         at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
>         at java.net.URLClassLoader.defineClass(URLClassLoader.java:473)
>         at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
>         at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
>         at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
>         at java.security.AccessController.doPrivileged(Native Method)
>         at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
>         at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
>         at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:151)
>         at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
>         at java.lang.Class.forName0(Native Method)
>         at java.lang.Class.forName(Class.java:348)
>         at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:46)
>         at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
>         at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
>         at org.springframework.boot.loader.WarLauncher.main(WarLauncher.java:59)

**Comment by MegaDrive68k** [comment]
Hi @Erlum 

I tried on Rocky Linux 8, but I was unsuccessful. I see that it uses your same OpenJDK version. Could you double-check that you're not using an alternative Java installation which could cause some conflict?

![Screenshot_2024-01-17_12-06-43](https://github.com/rundeck/rundeck/assets/51376003/71c1f184-6461-4458-9683-a66f397713b4)

![Screenshot_2024-01-17_12-07-12](https://github.com/rundeck/rundeck/assets/51376003/0b7905ab-9d37-47ab-9a5d-6538bffcdc19)

![Screenshot_2024-01-17_12-03-21](https://github.com/rundeck/rundeck/assets/51376003/82a1a456-1955-4e29-b734-a5ce75edd8e3)

**Comment by Erlum** [comment]
Before updating, I have the following packages :
```
root@RUNDECK rundeck# dnf info rundeck
Last metadata expiration check: 0:03:56 ago on Mon 22 Jan 2024 05:30:26 PM CET.
Installed Packages
Name         : rundeck
Version      : 4.17.3.20231113
Release      : 1
Architecture : noarch
Size         : 278 M
Source       : None
Repository   : @System
From repo    : rundeck
Summary      : Rundeck
URL          : http://rundeck.com
License      :
Description  : Rundeck

Available Packages
Name         : rundeck
Version      : 5.0.1.20240115
Release      : 1
Architecture : noarch
Size         : 265 M
Source       : None
Repository   : rundeck
Summary      : Rundeck
URL          : http://rundeck.com
License      :
Description  : Rundeck

root@RUNDECK rundeck# dnf list installed | grep java
java-1.8.0-openjdk.x86_64                1:1.8.0.392.b08-4.el8_8                 @appstream
java-1.8.0-openjdk-headless.x86_64       1:1.8.0.392.b08-4.el8_8                 @appstream
```

If I try to push the update, Rundeck goes to the next major and java follows with it. I'm using the packages available on the default repos (@Appstream) just like in your case. As you can see, this is not the exact same version though (11.0.21.0.9 vs 11.0.22.0.7) : 

```
root@RUNDECK rundeck# dnf update rundeck
rundeck                                                             271  B/s | 833  B     00:03
Dependencies resolved.
====================================================================================================
 Package                         Architecture  Version                       Repository        Size
====================================================================================================
Upgrading:
 rundeck                         noarch        5.0.1.20240115-1              rundeck          265 M
Installing dependencies:
 java-11-openjdk-headless        x86_64        1:11.0.22.0.7-2.el8           appstream         42 M

Transaction Summary
====================================================================================================
Install  1 Package
Upgrade  1 Package

Total download size: 306 M
Is this ok [y/N]:
```

**Comment by Erlum** [comment]
I'm going to close this issue as we're moving towards a licensed PDPA cluster and nobody else seeas to be affected. Thank you.

**Comment by ahormazabal** [comment]
According to the logs, you're still using a Java 8 runtime, but since Rundeck 5.0 the minimum JRE version 11: https://docs.rundeck.com/docs/history/5_x/version-5.0.0.html#java-11-minimum-required

Make sure that java 11 is the jre being actually used by the system to start the rundeck service.

**Comment by digdilem-work** [comment]
Had same issue on Rocky8. For others affected, the issue for me was resolved by removing the 1.8.0 packages after installing latest-openjdk packages

```
# dnf list installed |grep java
java-1.8.0-openjdk.x86_64                         1:1.8.0.422.b05-2.el8                                       @direct-appstream
java-1.8.0-openjdk-headless.x86_64                1:1.8.0.422.b05-2.el8                                       @direct-appstream
java-11-openjdk-headless.x86_64                   1:11.0.24.0.8-3.el8                                         @direct-appstream
java-latest-openjdk.x86_64                        1:23.0.0.0.37-1.rolling.el8                                 @direct-epel
java-latest-openjdk-headless.x86_64               1:23.0.0.0.37-1.rolling.el8                                 @direct-epel
javapackages-filesystem.noarch                    5.3.0-2.module+el8.3.0+125+5da1ae29                         @direct-appstream
tzdata-java.noarch                                2024a-1.el8                                                 @direct-appstream
```

``` 
dnf -y remove java-1.8.0-openjdk-headless.x86_64 java-1.8.0-openjdk.x86_64
dnf -y install java-latest-openjdk.x86_64
```
